nltk/twitter/util.py

```diff
@@ -0,0 +1,362 @@
+        outf = open(outfile, 'w', encoding=encoding, errors=errors)
+        writer = csv.writer(outf)
+    else:
+        outf = open(outfile, 'wb')
+        writer = compat.UnicodeWriter(outf, encoding=encoding, errors=errors)
+    return (writer, outf)
+
+
+
+
+def json2csv_entities(fp, outfile, main_fields, entity_name, entity_fields,
+                      encoding='utf8', errors='replace'):
+    """
+    Extract selected fields from a file of line-separated JSON tweets and
+    write to a file in CSV format.
+
+    This utility function allows a file of full tweets to be easily converted
+    to a CSV file for easier processing of Twitter entities. For example, the
+    hashtags or media elements of a tweet can be extracted.
+
+    :param file-object fp: The name of the file containing full tweets
+
+    :param str outfile: The name of the text file where results should be\
+    written
+
+    :param list main_fields: The list of fields to be extracted from the main\
+    object, usually the tweet. Useful examples: 'id_str' for the tweetID. See\
+    <https://dev.twitter.com/overview/api/tweets> for a full list of fields.
+    e. g.: ['id_str'], ['id', 'text', 'favorite_count', 'retweet_count']
+    If `entity_name` is expressed as a dictionary, then it is list of fields\
+    of the object that corresponds to the key of the dictionary (could be\
+    the user object, or the place of a tweet object).
+
+    :param list entity_name: The name of the entity: 'hashtags', 'media',\
+    'urls' and 'user_mentions' for the tweet object. For the user object,\
+    needs to be expressed as a dictionary: `{'user' : 'urls'}`. For the\
+    bounding box of the place from which a Tweet was published, as a dict\
+    as well: `{'place' : 'bounding_box'}`.
+
+    :param list entity_fields: The list of fields to be extracted from the\
+    entity. E.g. `['text']` (of the Tweet)
+
+    :param error: Behaviour for encoding errors, see\
+    https://docs.python.org/3/library/codecs.html#codec-base-classes
+    """
+
+    (writer, outf) = outf_writer_compat(outfile, encoding, errors)
+    for line in fp:
+        tweet = json.loads(line)
+        if isinstance(entity_name, dict):
+            for key, value in entity_name.items():
+                object_json = _get_entity_recursive(tweet, key)
+                if object_json == None:
+                    # can happen in the case of "place"
+                    continue
+                object_fields = extract_fields(object_json, main_fields)
+                items = _get_entity_recursive(object_json, value)
+                _write_to_file(object_fields, items, entity_fields, writer)
+        else:
+            tweet_fields = extract_fields(tweet, main_fields)
+            items = _get_entity_recursive(tweet, entity_name)
+            _write_to_file(tweet_fields, items, entity_fields, writer)
+    outf.close()
+
+
+def _write_to_file(object_fields, items, entity_fields, writer):
+    if items == None:
```

Comment:
PEP8 "Programming Recommendations" says:

> Comparisons to singletons like None should always be done with is or is not , never the equality operators. 

There is a couple more `== None` comparisons in this PR.